monkey/monkey_island/cc/services/authentication_service/flask_resources/agent_otp.py

```diff
@@ -12,12 +17,33 @@ class AgentOTP(AbstractResource):
     """
 
     urls = ["/api/agent-otp"]
+    lock = Lock()
+    limiter = None
+
+    def __init__(self, limiter: Limiter):
+        # Since flask generates a new instance of this class for each request,
+        # we need to ensure that a single instance of the limiter is used. Hence
+        # the class variable.
+        #
+        # TODO: The limit is currently applied per IP address. We will want to change
+        # it to per-user once we require authentication for this endpoint.
+        with AgentOTP.lock:
+            if AgentOTP.limiter is None:
+                AgentOTP.limiter = limiter.limit(
+                    "10/second", key_func=get_remote_address, per_method=True
+                )
 
     def get(self):
         """
         Requests an Agent's one-time password
 
         :return: One-time password in the response body
         """
-
-        return make_response({"otp": "supersecretpassword"})
+        if not AgentOTP.limiter:
+            raise RuntimeError("limiter has not been initialized")
```

Comment:
Is this possible? `__init__()` must be run before `get()` is called, or is that incorrect?

Reply:
It shouldn't be possible, but it satisfies `mypy`

Reply:
In that case, check `is None` instead of `not`. If Limiter objects are (or in the future, become) falsey, this check won't have the desired effect.